app/rust/src/ffi/call_request.rs

```diff
@@ -0,0 +1,196 @@
+/*******************************************************************************
+*   (c) 2024 Zondax AG
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+********************************************************************************/
+
+use crate::{
+    call_request::CallRequest, check_canary, constants::*, error::ParserError,
+    utils::ByteSerializable, FromBytes,
+};
+
+use core::mem::{size_of, MaybeUninit};
+use sha2::{Digest, Sha256};
+
+use super::resources::MEMORY_CALL_REQUEST;
+
+#[repr(C)]
+#[derive(PartialEq, Default)]
+#[cfg_attr(any(feature = "derive-debug", test), derive(Debug))]
+pub struct CanisterCallT {
+    pub arg_hash: [u8; 32],
+    pub canister_id: [u8; CANISTER_MAX_LEN],
+    pub canister_id_len: u16,
+    pub ingress_expiry: u64,
+    pub method_name: [u8; METHOD_MAX_LEN],
+    pub method_name_len: u16,
+    // pub request_type: [u8; REQUEST_MAX_LEN],
```

Comment:
Not that important, but should we clean this commented code ?

Reply:
indeed